lib/elixir/lib/string.ex

```diff
@@ -1812,4 +1812,91 @@ defmodule String do
       {comm + 1, trans, current}
     end
   end
+
+  @spec difference(t, t) :: [{:eq | :ins | :del, t}] | nil
+  def difference(str1, str2) do
```

Comment:
Any reason why we're not calling this `diff` instead of `difference`? `diff` may identify what this does in an easier way (as it's a known concept).